### PR TITLE
[CLB] Implemented Scion of Halaster

### PR DESCRIPTION
**Pull request by RJayz**
~Test with partner commanders is broken as both commanders gain the replacement effect and they keep triggering each other, will look into fixing this problem.~

Test with partners commanders has been fixed.

### CODE REVIEW

**Comment by Alex-Vasile** [comment]
Just an FYI, please comment on #8785 with any cards that you are currently working on so that others don't accidentally work on it as well.

File: Mage.Tests/src/test/java/org/mage/test/cards/single/clb/ScionOfHalasterTest.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+        playerA.getLibrary().clear();
+        addCard(Zone.LIBRARY, playerA, "Lightning Bolt");
+        addCard(Zone.LIBRARY, playerA, "Mountain");
+        skipInitShuffling();
+
+        // Continue to turn 3 so we get to Player A's draw step
+        setStopAt(3, PhaseStep.PRECOMBAT_MAIN);
+        execute();
+
+        // AI considers Lightning Bolt a better card than Mountain so it will discard the Mountain
```
Comment (Alex-Vasile) [inline Mage.Tests/src/test/java/org/mage/test/cards/single/clb/ScionOfHalasterTest.java:44]: If you use `setStrictChooseMode(true)` you can manually set which choices it should make.

File: Mage.Tests/src/test/java/org/mage/test/cards/single/clb/ScionOfHalasterTest.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+        assertLibraryCount(playerA, 0);
+        assertHandCount(playerA, 1);
+        assertHandCount(playerA, "Lightning Bolt", 1);
+        assertGraveyardCount(playerA, 1);
+        assertGraveyardCount(playerA, "Mountain", 1);
+    }
+
+    @Test
+    public void TestScionOfHalaster_withPartnerCommanders() {
```
Comment (Alex-Vasile) [inline Mage.Tests/src/test/java/org/mage/test/cards/single/clb/ScionOfHalasterTest.java:53]: Could you please include what the expected outcome of this test is as a docstring:
```Java
/**
*
*/
```

Comment (RJayz) [reply Mage.Tests/src/test/java/org/mage/test/cards/single/clb/ScionOfHalasterTest.java:53]: I've added an explanation, the test itself is not fixed yet

Comment (RJayz) [reply Mage.Tests/src/test/java/org/mage/test/cards/single/clb/ScionOfHalasterTest.java:62]: Test has been fixed

**Review by theelk801 — APPROVED** [review]

**Pull request merged**